src/c_api/c_api.cc

```diff
@@ -563,11 +587,21 @@ int MXLoadLib(const char *path) {
       }
 
       // create a pointer to hold custom op state object
+      // only create one stateful op depending on passing context
+      // user can add new supported context and call to custom library
       void* state_op_inst = nullptr;
-      CHECK(callCreateOpState(create_opstate_fp, attr_keys.data(), attr_vals.data(),
-                              attr_keys.size(), &state_op_inst))
-      << "Error calling CreateOpState for custom operator '" << name_str << "'";
-
+      if (ctx.dev_mask() == Context::kCPU) {
+        CHECK(createop_map.count("cpu") > 0) << "CPU CreateOpState not implemented";
```

Comment:
this is good, but can we made the error message more clear by adding the operator name at the end so the message is more like: CPU CreateOpState is not implemented for operator zzzzz

Reply:
agree